lstests/doc/README.md

```diff
@@ -0,0 +1,692 @@
+```
+
+the user may create a cluster with a specific genesis file.
+
+```python
+clus = Cluster(service=service, extra_args="--genesis-json=gene.json")
+```
+
+This specifies `gene.json` as the genesis file for the cluster, which overrides the service-level setting.
+
+## Testing
+
+The `Cluster` class provides a list of methods, such as `set_contract()` and `push_actions()`, which serve as building blocks for a test script to build up the testing scenario.
+
+### Mechanism
+
+Behind the scenes, a `Cluster` object communicates with the launcher service via HTTP connection in a request-and-response model.
+
+Almost all the methods finally passes through or are dependent on the `call()`  method in the `Cluster` class. The standard steps to make a call include
+
+```
+  Steps to take to make a call
+  ----------------------------
+  1. print header
+  2. establish HTTP connection
+  3. log URL and request of connection
+  4. retry connection if response not ok
+  5. log response
+  6. verify transaction
+  7. return the result as a Connection object
+```
+
+The exact actions to take are fully customizable. Refer to `**call_kwargs` for the full control.
+
+### Main API
+
+After the node cluster is successfully launched, all the test actions can be performed on the cluster, with main API listed below.
+
+    Main API
+    --------
+    - Start up and shut down
+        - launch_cluster()
+        - stop_cluster()
+        - start_node()
+        - stop_node()
+        - stop_all_nodes()
+    - Queries
+        - get_cluster_running_state()
+        - get_cluster_info()
+        - get_info()
+        - get_block()
+        - get_account()
+        - get_protocol_features()
+        - get_log()
+    - Transactions
+        - schedule_protocol_feature_activations()
+        - set_contract()
+        - push_action()
+    - Check and Verify
+        - verify()
+        - check_sync()
+        - check_production_round()
+    - Miscellaneous
+        - send_raw()
+        - pause_node_production()
+        - resume_node_production()
+        - get_greylist()
+        - add_greylist_accounts()
+        - remove_greylist_accounts()
+        - get_net_plugin_connections()
+The full specification of the methods can be found in their docstrings.
+
+### `**call_kwargs`
+
+One special parameter across almost all the methods is `**call_kwargs`, which represents all the controlling keyword arguments in the `call` method, which include
+
+```
+ retry
+ sleep
+ dont_raise
+ verify_async
+ verify_key
+ verify_sleep
+ verify_retry
+ verify_assert
+ verify_dont_raise
+ header
+ level
+ header_level
+ url_level
+ request_text_level
+ retry_info_level
+ retry_text_level
+ response_code_level
+ response_text_level
+ transaction_id_level
+ no_transaction_id_level
+ error_level
+ error_text_level
+ verify_level
+ verify_retry_level
+ verify_error_level
+ buffer
+ dont_flush
+```
+
+These parameters adds to the flexibility of a test script. Consider
```

Comment:
"These parameters adds" -> "These parameters add"

Reply:
Fixed. Thank you.